app/Gists/File.php

```diff
@@ -0,0 +1,64 @@
+<?php namespace Gistlog\Gists;
+
+use Illuminate\Contracts\Support\Arrayable;
+
+class File implements Arrayable
+{
+    public $name;
+    public $type;
+    public $language;
+    public $url;
+    public $size;
+    public $content;
+
+    /**
+     * @param array|ArrayAccess $rawFile
+     * @return File
+     */
+    public static function fromGitHub($rawFile)
+    {
+        $file = new self;
+
+        $file->name = $rawFile['filename'];
+        $file->type = $rawFile['type'];
+        $file->language = $rawFile['language'];
+        $file->url = $rawFile['raw_url'];
+        $file->size = $rawFile['size'];
+        $file->content = $rawFile['content'];
+
+        return $file;
+    }
+
+    /**
+     * @param array|ArrayAccess $rawFiles
+     * @return Collection
+     */
+    public static function multipleFromGitHub($rawFiles)
+    {
+        // Remove the blog post itself.
+        array_shift($rawFiles);
```

Comment:
Only thing I'm unsure of is whether I'm comfortable with assuming the blogpost is the first. I think I remember that the order that comes through is alphabetical by filename; so, for example, if they used `blog.md`, which I think I may have recommended, and they have `admin.css`, this whole thing will fall apart.

@adamwathan @JacobBennett y'all remember.. am I right in remembering how this works out?

I'm trying to think about good solutions, and so far I've come up with just suggesting folks prefix +blog.md+ with an underscore or something, or having them define which file is the blog file in the YAML config (don't love this one.)